### PR TITLE
Update CNI to the 0.9 release

### PR DESCRIPTION
**Pull request by lachie83**
This change updates the following:

- Azure CNI plugin to 0.9
- Introduces new vnetCidr key to exclude from the network masquerade block

This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1259)

### CODE REVIEW

File: parts/kubernetesmastercustomscript.sh [hunk 1]
```diff
@@ -186,7 +186,7 @@ function configAzureNetworkPolicy() {
     mkdir -p $CNI_BIN_DIR
 
     # Mirror from https://github.com/Azure/azure-container-networking/releases/tag/$AZURE_PLUGIN_VER/azure-vnet-cni-linux-amd64-$AZURE_PLUGIN_VER.tgz
-    downloadUrl https://acs-mirror.azureedge.net/cni/azure-vnet-cni-linux-amd64-latest.tgz | tar -xz -C $CNI_BIN_DIR
+    downloadUrl https://github.com/ofiliz/hello/releases/download/pre-release/azure-vnet-cni-linux-amd64-v0.8-7-g8acef7d.tgz | tar -xz -C $CNI_BIN_DIR
```
Comment (jackfrancis) [inline parts/kubernetesmastercustomscript.sh]: I don't think we don't want to merge this to master until we test/vet the external `azure-vnet-cni-linux-amd64` download and then host it in `https://acs-mirror.azureedge.net/cni`.

File: parts/kubernetesmastervars.t [hunk 2]
```diff
@@ -102,6 +102,7 @@
     "subnetName": "[split(parameters('masterVnetSubnetID'), '/')[variables('subnetNameResourceSegmentIndex')]]",
     "vnetNameResourceSegmentIndex": 8,
     "virtualNetworkName": "[split(parameters('masterVnetSubnetID'), '/')[variables('vnetNameResourceSegmentIndex')]]",
+    "vnetCidr": "10.0.0.0/8",
```
Comment (lachie83) [inline parts/kubernetesmastervars.t]: I will make this configurable in the api model

File: parts/kubernetesmastercustomscript.sh (hunk 1 above)
Comment (lachie83) [inline parts/kubernetesmastercustomscript.sh]: Correct!

File: pkg/acsengine/defaults.go [hunk 3]
```diff
@@ -158,6 +158,7 @@ func setMasterNetworkDefaults(a *api.Properties) {
 				// When VNET integration is enabled, all masters, agents and pods share the same large subnet.
 				a.MasterProfile.Subnet = a.OrchestratorProfile.KubernetesConfig.ClusterSubnet
 				a.MasterProfile.FirstConsecutiveStaticIP = getFirstConsecutiveStaticIPAddress(a.MasterProfile.Subnet)
+				a.MasterProfile.VnetCidr = (a.MasterProfile.VnetCidr)
```
Comment (lachie83) [inline pkg/acsengine/defaults.go]: @jackfrancis is this required given that I set a default for the param in kubernetesparams.t

Comment (jackfrancis) [inline pkg/acsengine/defaults.go]: Yeah, I'm not sure what that statement will accomplish. Do you get the expected behavior in both cases (default and user-provide) without that statement?

Comment (lachie83) [inline pkg/acsengine/defaults.go]: Yes. I've yanked this

**Pull request merged**